src/test/coll.rs

```diff
@@ -1020,3 +1020,28 @@ async fn drop_skip_serializing_none() {
     let options = DropCollectionOptions::builder().build();
     assert!(coll.drop(options).await.is_ok());
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn collection_generic_bounds() {
```

Comment:
this test and the other one need to acquire the test lock

Reply:
added the lock to the async test (and to another async test that wasn't using it), we weren't using the lock at all in the sync tests so I updated all of those as well